lua/lazy/manage/rocks.lua

```diff
@@ -0,0 +1,90 @@
+--# selene:allow(incorrect_standard_library_use)
+
+local Config = require("lazy.core.config")
+local Util = require("lazy.core.util")
+
+---@class LazyRock
+---@field plugin string
+---@field name string
+---@field spec string
+---@field installed boolean
+
+local M = {}
+---@type LazyRock[]
+M.rocks = {}
+
+---@param ... string
+---@return string[]
+function M.args(...)
+  local ret = {
+    "--tree",
+    Config.rocks.tree,
+    "--server",
+    Config.options.rocks.server,
+    "--dev",
+    "--lua-version",
+    "5.1",
+  }
+  vim.list_extend(ret, { ... })
+  return ret
+end
+
+---@param plugin LazyPlugin
+function M.get_rockspec(plugin)
```

Comment:
as far as I can tell this is still from what was done in https://github.com/folke/lazy.nvim/pull/1528 where users would have to add dependencies with `rocks = {}`
imo if rockspec is supported this shouldn't be added because it will encourage plugin authors to use this instead of a rockspec which has the issue of causing breaking changes when dependencies change

Reply:
With my latest changes this is now indeed no longer needed.
Added proper support for building luarocks packages.

You can now also just add any luarocks package to your config like:

```lua
{ "https://github.com/lubyk/yaml" }
```

This will properly install/build the rock and make it available inside Neovim